omod/src/main/java/org/openmrs/web/controller/ForgotPasswordFormController.java

```diff
@@ -119,16 +121,23 @@ protected ModelAndView onSubmit(HttpServletRequest request, HttpServletResponse
 					if (username != null && username.length() > 0) {
 						user = Context.getUserService().getUserByUsername(username);
 					}
-				}
-				finally {
+				} finally {
 					Context.removeProxyPrivilege(PrivilegeConstants.GET_USERS);
 				}
 				
-				if (user == null /*|| StringUtils.isEmpty(user.getSecretQuestion())*/) {
+				String secretQuestion = null;
+				if (user != null) {
+					secretQuestion = Context.getUserService().getSecretQuestion(user);
+				}
+				
+				if (user == null) {
```

Comment:
Yup I have done it. I'll incorporate all the changes you ask me to and create a new pull request asap